clang/test/CodeGenSYCL/unique_stable_name_windows_diff.cpp

```diff
@@ -3,17 +3,17 @@
 
 
 template<typename KN, typename Func>
-__attribute__((sycl_kernel)) void kernel(Func F){
+__attribute__((sycl_kernel_entry_point(KN))) void kernel(Func F){
   F();
 }
 
 template<typename KN, typename Func>
-__attribute__((sycl_kernel)) void kernel2(Func F){
+__attribute__((sycl_kernel_entry_point(KN))) void kernel2(Func F){
   F(1);
 }
 
 template<typename KN, typename Func>
-__attribute__((sycl_kernel)) void kernel3(Func F){
+__attribute__((sycl_kernel_entry_point(KN))) void kernel3(Func F){
```

Comment:
Looking more closely at this, I don't think we should make this change. The new design doesn't use the `__builtin_sycl_unique_stable_name()` attribute. The fact that the test doesn't break with this change is interesting and surprising; I would expect it to since the new design uses differently mangled names.

Reply:
I am confused. The output being tested is the output of the builtin right? The mangling for the builtin has not changed.

Reply:
If the test was intended to only test the output of the builtin, then there would be no need to declare and call any SYCL entry point functions. It seems that 1) the test should either validate that the emitted SYCL kernel caller functions have a name that matches what the calls to the builtin produce, or 2) that lines 5-18 and 28-30 should be removed.

Reply:
>>then there would be no need to declare and call any SYCL entry point functions
This is not entirely true. In `unique_stable_name.cpp` , `the sycl_kernel_entry_point` attribute is required to emit the actual code in device compilation. The builtin is called within the kernel. For cases where the builtin is not called within a kernel, host compilation is run and those do not require the entry point functions. 

For `unique_stable_name_windows_diff.cpp`, I just noticed that the builtin is called in host code. We still emit the IR because of the hack we have in place to unconditionally emit code in device for the moment but this is wrong. To fix this test properly, we need to call the builtins in device code as is done in syclos now using `kernel_wrapper` I think. 

Thinking about it now, I am not sure how to handle this builtin going forward. Do we plan to remove it? Are there other SYCL implementations using this? Is it used to get a unique name for any type i.e. not just kernel names? I don't know how this builtin is implemented but it looks like the output does not depend on whether the argument is an actual kernel name or not, considering no underlying code for this builtin have been changed to use the new attribute.

Reply:
I've modified `unique_stable_name_windows_diff.cpp` to call the builtins in device code as it should be. This solution is equivalent to what we have in syclos at the moment.

Reply:
I can try removing the kernel calls as you suggested and see if that changes anything. I don't expect it to.

Reply:
Test passes even if I remove the kernel calls.